### PR TITLE
Fixed response

### PR DESCRIPTION
**Pull request by zigsas**

### CODE REVIEW

File: backend/WebApplication1/Controllers/DefaultController.cs [hunk 1]
```diff
@@ -29,21 +31,33 @@ public DefaultController(PlateService plateService, FaceService faceService, ILo
         [ProducesResponseType(400)]
         public async Task<IActionResult> PostAsync([FromBody] string imageBase64)
         {
-            _logger.Log(LogType.NORMAL, "POST Request started in PostAsync()\n");
+            _logger.Log(LogType.NORMAL, "POST Request started in PostAsync()");
             try
```
[the remark this answers is not in the document]
Comment (tozaicevas) [reply backend/WebApplication1/Controllers/DefaultController.cs]: agreed, keep logging logic outside of try catch, the exception can occur only in these lines:
`                Task<List<RecognizedObject>> getPlateResponseTask = _plateService.RecognizeAsync(imageBase64);
                Task<List<RecognizedObject>> getPersonResponseTask = _faceService.RecognizeAsync(imageBase64);`

File: backend/WebApplication1/Models/Responses/RecognitionResponse.cs [hunk 2]
```diff
@@ -1,13 +1,11 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Threading.Tasks;
 
 namespace WebApplication1.Models.Responses
 {
-    public struct PlateResponse
+    public struct RecognitionResponse
```
Comment (tozaicevas) [inline backend/WebApplication1/Models/Responses/RecognitionResponse.cs]: where is this struct being used?

Comment (UndeadRat22) [inline backend/WebApplication1/Models/Responses/RecognitionResponse.cs]: Unused struct?

File: backend/WebApplication1/Models/Responses/RecognizedObject.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+using static WebApplication1.Models.Abstract.MissingModel;
+
+namespace WebApplication1.Models.Responses
+{
+    public struct RecognizedObject
+    {
+        public string FirstName { get; set; }
+        public string LastName { get; set; }
+        public SearchReason Reason { get; set; }
+        public string Type { get; set; }
```
Comment (tozaicevas) [inline backend/WebApplication1/Models/Responses/RecognizedObject.cs]: make it an enum and just return it, frontend will map it to a specific type

Comment (tozaicevas) [reply backend/WebApplication1/Models/Responses/RecognizedObject.cs]: just like Reason

File: backend/WebApplication1/Services/PlateService.cs [hunk 4]
```diff
@@ -1,77 +1,75 @@
...
         private PlateRepository _plateRepository;
+        private TimestampRepository _timestampRepository;
         private readonly ITimestamper<Plate> _timestamper;
 
-        public PlateService(PlateRepository plateRepository, ITimestamper<Plate> timestamper)
+        public PlateService(PlateRepository plateRepository, TimestampRepository timestampRepository, ITimestamper<Plate> timestamper)
         {
             _plateRepository = plateRepository;
+            _timestampRepository = timestampRepository;
             _timestamper = timestamper;
```
Comment (tozaicevas) [inline backend/WebApplication1/Services/PlateService.cs]: this is no good, we shouldn't need 2 objects for same-domain tasks, timestampRepository should handle both saving and retrieving timestamps

File: backend/WebApplication1/Services/PlateService.cs [hunk 5]
```diff
@@ -1,77 +1,75 @@
...
+                            DateAndTime = DateTime.Now.GetFormattedDateAndTime(),
+                            PlateID = plate.ID
+                        };
+                    }
+                    identifiedPlates.Add(new RecognizedObject()
+                    {
+                        FirstName = plate.FirstName,
+                        LastName = plate.LastName,
+                        Reason = plate.Reason,
+                        Type = "Plate",
```
Comment (tozaicevas) [inline backend/WebApplication1/Services/PlateService.cs]: pass enum type

File: backend/WebApplication1/Services/PlateService.cs (hunk 4 above)
Comment (UndeadRat22) [reply backend/WebApplication1/Services/PlateService.cs]: nice catch.

File: backend/WebApplication1/Controllers/DefaultController.cs [hunk 6]
```diff
@@ -32,34 +32,33 @@ public DefaultController(PlateService plateService, FaceService faceService, ILo
...
             {
                 _logger.Log(LogType.ERROR, ex.Message);
                 return BadRequest(ex.Message);
             }
+            List<RecognizedObject> plateResponse = await getPlateResponseTask;
+            List<RecognizedObject> personResponse = await getPersonResponseTask;
+            RecognizedObject[] responses = plateResponse.Concat(personResponse).ToArray();
+            if (responses.Length > 0)
+            {
+                _logger.Log(LogType.NORMAL, MessageBuilder.BuildResponseMessage(responses));
```
Comment (tozaicevas) [inline backend/WebApplication1/Controllers/DefaultController.cs]: Well done!

**Pull request merged**